fineract-provider/src/main/java/org/apache/fineract/accounting/closure/service/CalculateIncomeAndExpenseBookingImpl.java

```diff
@@ -0,0 +1,187 @@
+                if(incomeAndExpense.getOfficeRunningBalance().signum() == 1){
+                    debits = debits.add(incomeAndExpense.getOfficeRunningBalance());
+                    debitsJournalEntry.add(new SingleDebitOrCreditEntryData(incomeAndExpense.getAccountId(),incomeAndExpense.getGlAccountName(),incomeAndExpense.getOfficeRunningBalance(),null));
+                }else{
+                    credits= credits.add(incomeAndExpense.getOfficeRunningBalance().abs());
+                    creditsJournalEntry.add(new SingleDebitOrCreditEntryData(incomeAndExpense.getAccountId(),incomeAndExpense.getGlAccountName(),incomeAndExpense.getOfficeRunningBalance().abs(),null));;
+                }
+            }
+            if(incomeAndExpense.isExpenseAccountType()){
+                if(incomeAndExpense.getOfficeRunningBalance().signum() == 1){
+                    credits = credits.add(incomeAndExpense.getOfficeRunningBalance());
+                    creditsJournalEntry.add(new SingleDebitOrCreditEntryData(incomeAndExpense.getAccountId(),incomeAndExpense.getGlAccountName(),incomeAndExpense.getOfficeRunningBalance().abs(),null));;
+                }else{
+                    debits= debits.add(incomeAndExpense.getOfficeRunningBalance().abs());
+                    debitsJournalEntry.add(new SingleDebitOrCreditEntryData(incomeAndExpense.getAccountId(),incomeAndExpense.getGlAccountName(),incomeAndExpense.getOfficeRunningBalance().abs(),null));
+                }
+            }
+        }
+        final LocalDate today = DateUtils.getLocalDateOfTenant();
+        final int compare = debits.compareTo(credits);
+        BigDecimal difference = BigDecimal.ZERO;
+        JournalEntryData journalEntry = null;
+        if(compare == 1){
+            /* book with target gl id on the credit side */
+            difference = debits.subtract(credits);
+            SingleDebitOrCreditEntryData targetBooking = new SingleDebitOrCreditEntryData(closureData.getEquityGlAccountId(),glAccount.getName(),difference,null);
+            creditsJournalEntry.add(targetBooking);
+            journalEntry = new JournalEntryData(office.getId(),today.toString(),closureData.getComments(),creditsJournalEntry,debitsJournalEntry,null,false,closureData.getCurrencyCode(),office.getName());
+        }else if(compare == -1){
+            /* book with target gl id on the debit side*/
+            difference = credits.subtract(debits);
+            SingleDebitOrCreditEntryData targetBooking = new SingleDebitOrCreditEntryData(closureData.getEquityGlAccountId(),glAccount.getName(),difference,null);
+            debitsJournalEntry.add(targetBooking);
+            journalEntry = new JournalEntryData(office.getId(),today.toString(),closureData.getComments(),creditsJournalEntry,debitsJournalEntry,null,false,closureData.getCurrencyCode(),office.getName());
+        }
+        else if(compare == 0){
+            //throw new RunningBalanceZeroException(office.getName());
```

Comment:
?

Reply:
https://github.com/muellners/fineract/blob/38e19f73fcea5c9165d34d9c83a7be8bf6cec884/fineract-provider/src/main/java/org/apache/fineract/accounting/closure/service/GLClosureWritePlatformServiceJpaRepositoryImpl.java#L353

Kinda of strange , in my repo I updated this, but does it not reflect here,

Reply:
You are looking at the old revision @vorburger